src/main/java/dk/alexandra/fresco/lib/math/logic/LogicProtocolBuilder.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * 
+ */
+package dk.alexandra.fresco.lib.math.logic;
+
+import java.math.BigInteger;
+
+import dk.alexandra.fresco.framework.ProtocolProducer;
+import dk.alexandra.fresco.framework.value.SInt;
+import dk.alexandra.fresco.lib.field.integer.BasicNumericFactory;
+import dk.alexandra.fresco.lib.helper.builder.AbstractProtocolBuilder;
+
+/**
+ * @author mortenvchristiansen
+ *
+ */
+public class LogicProtocolBuilder extends AbstractProtocolBuilder{
+
+	
+	private final BasicNumericFactory bnf;
+	
+	private final SInt trueSInt;
+	private final SInt falseSInt;
+		
+	@SuppressWarnings("deprecation")
+	public LogicProtocolBuilder(BasicNumericFactory bnf) {
+		this.bnf=bnf;
+		trueSInt = bnf.getSInt(BigInteger.ONE);
```

Comment:
We're so sorry that we have not fixed this yet, but this is not how values should be loaded. It should be with the method public KnownSIntProtocol getSInt(BigInteger i, SInt si);
I know this is annoying, but it's for sure more correct. This means first loading SInt values at the evaluation step. The naive fix for this is to call the method each time you need trueSInt or falseSInt.. Smarter approaches are sure to exist.